src/DataGrid.tsx

```diff
@@ -996,8 +996,17 @@ function DataGrid<R, SR, K extends Key>(
 
     const onRowChange = (row: R, commitChanges?: boolean) => {
       if (commitChanges) {
-        updateRow(column, selectedPosition.rowIdx, row);
-        closeEditor();
+        // Fixes two issues when editor is closed by clicking on a different cell
+        //
+        // Sometimes commitEditorChanges is called before the cell state is changed to
+        // SELECT and this results in onRowChange getting called twice.
+        //
+        // Sometimes rows prop is changed before selectCell is called and this results in
+        // rows.indexOf(row) returning -1 so incorrect cell gets selected
```

Comment:
```suggestion
        // rows.indexOf(row) returning -1 so no cell can be selected
```

Reply:
It does select the header or summary row if present